.ci/test.sh

```diff
@@ -103,6 +103,7 @@ if [[ $TASK == "lint" ]]; then
         'mypy>=1.11.1' \
         'pre-commit>=3.8.0' \
         'pyarrow-core>=17.0' \
+        'scikit-learn>=1.15.0' \
```

Comment:
This is to ensure that `mypy` checks `scikit-learn` imports. Extra important now that I'm proposing adding an optional type hint on this new `sklearn.utils.Tags`.